coalib/nestedlib/NlFileHandler.py

```diff
@@ -0,0 +1,179 @@
+    """
+    file = File(orig_file_path)
+
+    # Initialiaze a line_list with whitespace. The lenght of this list will
+    # be equal to the number of lines in the actual files.
+    # The index of the list points to the line number of the original file.
+    # The means line_list[3] contains the line at line number 4 of the original
+    # file.
+    line_list = []
+    line_list = [' ' for index in range(0, file.__len__())]
+
+    for nl_section in nl_sections:
+
+        start_line = nl_section.start.line
+        start_column = nl_section.start.column
+        end_line = nl_section.end.line
+        end_column = nl_section.end.column
+
+        for line_nr in range(start_line, end_line+1):
+
+            # Make the length of line equal to the length of original line
+            if(line_list[line_nr-1].isspace()):
+                line_list[line_nr-1] = ' '*len(file[line_nr-1])
+
+            orig_line = file[line_nr-1]
+            end_orig_line = len(orig_line)-1
+            line = line_list[line_nr-1]
+
+            # If the section contains only one line.
+            # This case generally happens for mixed lang line
+            if (line_nr == start_line and line_nr == end_line):
+                # section_content stores the part of the line that belongs to
+                # the section.
+                section_content = file[line_nr-1][start_column-1:end_column]
+
+                if(start_column-1 > 0) and (end_column < end_orig_line):
+                    line_list[line_nr-1] = (line[0:start_column-1] +
+                                            section_content +
+                                            line[end_column:end_orig_line])
+
+                elif(start_column-1 == 0) and (end_column < end_orig_line):
+                    line_list[line_nr-1] = (section_content +
+                                            line[end_column:end_orig_line])
+
+                elif(start_column-1 > 0) and (end_column == end_orig_line):
+                    line_list[line_nr-1] = (line[0:start_column-1] +
+                                            section_content)
+                else:
+                    line_list[line_nr-1] = section_content
+
+            elif (line_nr == start_line):
+                if(start_column-1 == 0):
+                    line_list[line_nr-1] = orig_line
+                else:
+                    line_list[line_nr-1] = (line[0:start_column-1] +
+                                        orig_line[start_column-1:end_orig_line])
+
+            elif (line_nr == end_line):
+                if(end_column == len(orig_line)-1):
+                    line_list[line_nr-1] = orig_line
+                else:
+                    line_list[line_nr-1] = (orig_line[0:end_column-1] +
+                                            line[end_column-1:end_orig_line])
+
+            else:
+                line_list[line_nr-1] = orig_line
+    return line_list
+
+
+def beautify_line_list(line_list):
```

Comment:
Do you anticipate any issues while using this functionality to recombine the parts of files?
I see the immediate utility, but modifying the files means we are doing additional linting apart from what the user indicated (in the form of bears), which may / may not be desired

Do correct me if I got this wrong

Reply:
Nope, I do not anticipate any issues with this functionality. The reason being, we will be maintaining the list of `NlSection`, which contain the `start_line`, `start_column`, `end_line` and `end_column`. 

And the only thing we do in `beautify_line_list` is add `\n`, in order to make the temporary files valid. Note that `\n` is only added where we have `empty` entry in the list, These empty entries actually represent the lines of the files of another programming language. If we do not do this, the temporary files formed will be invalid.